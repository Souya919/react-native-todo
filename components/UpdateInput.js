import React, { useState, useEffect } from "react";
import { View, TextInput, TouchableOpacity } from "react-native";

import { AntDesign } from "@expo/vector-icons";

export default function UpdateInput({ selectedItem, updateItem }) {
  const [value, setValue] = useState("");

  useEffect(() => {
    // Set the value state with the selected item's value whenever it changes
    setValue(selectedItem.value);
  }, [selectedItem]);

  // Updates the value state with the input text
  const updateValue = (val) => {
    setValue(val);
  };

  const updateItemOnSubmit = () => {
    // If a value exists, call the updateItem function with the selected item's key and the updated value
    if (value) updateItem(selectedItem.key, value);
  };

  return (
    <View style={styles.componentContainer}>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          placeholder="Update Task..."
          value={value}
          onChangeText={updateValue}
        />
      </View>
      <TouchableOpacity
        style={styles.submitButton}
        onPress={updateItemOnSubmit}
      >
        <AntDesign name="edit" size={24} color="midnightblue" />
      </TouchableOpacity>
    </View>
  );
}

const styles = {
  componentContainer: {
    flexDirection: "row",
  },
  inputContainer: {
    flexDirection: "row",
    borderRadius: 10,
  },
  input: {
    fontFamily: "poppins-regular",
    fontSize: 20,
    backgroundColor: "white",
    width: 300,
    marginRight: 20,
    padding: 10,
    marginBottom: 20,
    borderRadius: 10,
  },
  submitButton: {
    width: 50,
    justifyContent: "center",
    alignItems: "center",
    backgroundColor: "whitesmoke",
    marginBottom: 20,
    borderRadius: 50,
  },
};
