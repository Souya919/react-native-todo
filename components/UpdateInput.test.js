import React from "react";
import renderer from "react-test-renderer";
import UpdateInput from "./UpdateInput";

// Mock the necessary dependencies
jest.mock("@expo/vector-icons", () => ({
  AntDesign: "AntDesign",
}));

describe("<UpdateInput />", () => {
  it("renders correctly", async () => {
    // Define mock updateItem function
    const updateItem = jest.fn();

    // Render the UpdateInput component with the required prop
    let tree;

    await renderer.act(async () => {
      tree = renderer.create(
        <UpdateInput
          selectedItem={{ key: "1", value: "Task" }}
          updateItem={updateItem}
        />
      );
    });

    expect(tree).toMatchSnapshot();
  });
});
