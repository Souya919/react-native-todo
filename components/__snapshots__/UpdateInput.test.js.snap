// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UpdateInput /> renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
    }
  }
>
  <View
    style={
      {
        "borderRadius": 10,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Update Task..."
      style={
        {
          "backgroundColor": "white",
          "borderRadius": 10,
          "fontFamily": "poppins-regular",
          "fontSize": 20,
          "marginBottom": 20,
          "marginRight": 20,
          "padding": 10,
          "width": 300,
        }
      }
      value="Task"
    />
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "whitesmoke",
        "borderRadius": 50,
        "justifyContent": "center",
        "marginBottom": 20,
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <AntDesign
      color="midnightblue"
      name="edit"
      size={24}
    />
  </View>
</View>
`;
